src/DraupMembershipERC721.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {IERC2981, IERC165} from "openzeppelin-contracts/contracts/interfaces/IERC2981.sol";
+import {DefaultOperatorFilterer} from "operator-filter-registry/src/DefaultOperatorFilterer.sol";
+import {ERC721} from "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {MerkleProof} from "openzeppelin-contracts/contracts/utils/cryptography/MerkleProof.sol";
+import {PaddedString} from "draup-utils/src/PaddedString.sol";
+import {IRenderer} from "./IRenderer.sol";
+
+contract DraupMembershipERC721 is ERC721, Ownable, DefaultOperatorFilterer {
+    uint256 public immutable MAX_SUPPLY;
+    uint256 public immutable ROYALTY = 750;
+    bool public transfersAllowed = false;
+    IRenderer public renderer;
+    string public baseTokenURI;
+    uint256 public nextTokenId;
+    bytes32 public merkleRoot;
+
+    // Mapping to track who used their allowlist spot
+    mapping(address => bool) private _claimed;
+
+    constructor(uint256 maxSupply, string memory baseURI) ERC721("Draup Membership", "DRAUP") {
+        MAX_SUPPLY = maxSupply;
+        baseTokenURI = baseURI;
+    }
+
+    error InvalidProof();
+    error AlreadyClaimed();
+    error MaxSupplyReached();
+    error TransfersNotAllowed();
+
+    function mint(bytes32[] calldata merkleProof) public {
```

Comment:
I think it's generally nice to have a `mintTo` function where you can specify a recipient address. ie

```
function mintTo(address to, bytes32[] calldata merkleProof) public;
```

This allows flexibility in how the mint occurs. Some benefits:

1. Can mint gaslessly with relayers/meta-transactions
2. Could mint from a hot wallet into a cold wallet
3. .. other things?

Of course, if you do this, you need to change logic around `msg.sender`. What addresses go into the merkle tree? The minters or the recipients of mints? Probably better to verify the recipient is in the tree, that way relayers could mint on behalf of recipients, etc.

Not sure if it's worth it to you to add a recipient address!

Reply:
You know what, you would definitely want to do merkle proof checks on the recipient if you had a `mintTo`. Otherwise if it was on a `msg.sender` that could be different than the recipient, I could see your in flight transaction, get the merkle proof, and then submit a different recipient.